test/infrastructure/container/docker.go

```diff
@@ -505,7 +505,13 @@ func (d *dockerRuntime) RunContainer(ctx context.Context, runConfig *RunContaine
 
 	// Actually start the container
 	if err := d.dockerClient.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
-		return errors.Wrapf(err, "error starting container %q", runConfig.Name)
+		err := errors.Wrapf(err, "error starting container %q", runConfig.Name)
+		// Delete the container and retry later on. This helps getting around the race
+		// condition where of hitting "port is already allocated" issues.
+		if innerErr := d.dockerClient.ContainerRemove(ctx, resp.ID, types.ContainerRemoveOptions{Force: true, RemoveVolumes: true}); innerErr != nil {
+			return errors.Wrapf(innerErr, "error removing container after failed start: %s", err)
```

Comment:
Not sure if that mixes the errors in a way that is not easily readable. I think ideally we would use kerrors aggregate (and usually do)

Reply:
In this case it should result in:
```
{InnerErr}: error removing container after failed start: {err}
```

Ack, kerrors aggregate would have been an option I didn't think of.

Was also thinking about using go's `errors.Join`.

Reply:
I think this is not how error wrapping works. As far as I know it appends the innerErr at the end

Reply:
(just like the normal fmt.Errorf("adfasfd test: %v", err) would)

Reply:
Thx! I know it's a nit in a test provider, just noticed and was thinking about future me trying to parse the error :)